sdk/identity/Azure.Identity/tests/KubernetesProxyHttpHandlerTests.cs

```diff
@@ -0,0 +1,1228 @@
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.IsTrue(response.Headers.Contains("X-Custom-Response"));
+            Assert.IsTrue(response.Content.Headers.Contains("Content-Type"));
+        }
+
+        [Test]
+        public async Task SendAsync_HandlesNon200Responses()
+        {
+            // Arrange
+            var mockTransport = new MockTransport(req =>
+            {
+                var response = new MockResponse(401);
+                response.SetContent("{\"error\":\"unauthorized\"}");
+                return response;
+            });
+
+            var proxyConfig = CreateTestConfig("https://proxy.example.com", "proxy.sni.example.com", transport: mockTransport);
+
+            var proxyHandler = new KubernetesProxyHttpHandler(proxyConfig);
+            var httpClient = new HttpClient(proxyHandler);
+
+            // Act
+            var response = await httpClient.GetAsync("https://login.microsoftonline.com/token");
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.Unauthorized, response.StatusCode);
+        }
+
+        [Test]
+        public async Task SendAsync_DoesNotReloadHandlerWhenNoCaFile()
+        {
+            // Arrange
+            var requestCount = 0;
+            var mockTransport = new MockTransport(req =>
+            {
+                requestCount++;
+                var response = new MockResponse(200);
+                response.SetContent($"{{\"request\":{requestCount}}}");
+                return response;
+            });
+
+            var proxyConfig = CreateTestConfig("https://proxy.example.com", "proxy.sni.example.com", transport: mockTransport);
+
+            var proxyHandler = new KubernetesProxyHttpHandler(proxyConfig);
+            var httpClient = new HttpClient(proxyHandler);
+
+            // Act - Make multiple requests
+            await httpClient.GetAsync("https://login.microsoftonline.com/token");
+            await httpClient.GetAsync("https://login.microsoftonline.com/token");
+            await httpClient.GetAsync("https://login.microsoftonline.com/token");
+
+            // Assert
+            Assert.AreEqual(3, requestCount);
+        }
+
+        [Test]
+        public async Task SendAsync_DoesNotReloadHandlerWhenCaFileIsEmpty()
+        {
+            // Arrange
+            var caFilePath = _tempFiles.GetTempFilePath();
+            File.WriteAllText(caFilePath, ""); // Empty file
+
+            var requestCount = 0;
+            var mockTransport = new MockTransport(req =>
+            {
+                requestCount++;
+                var response = new MockResponse(200);
+                response.SetContent($"{{\"request\":{requestCount}}}");
+                return response;
+            });
+
+            var proxyConfig = CreateTestConfig("https://proxy.example.com", "proxy.sni.example.com", caFilePath, transport: mockTransport);
+
+            var proxyHandler = new KubernetesProxyHttpHandler(proxyConfig);
```

Comment:
Shouldn't this or the gets throw? The configuration is invalid because it specifies an empty CA file